---
title: ""
subtitle: "Cell assignation report"
author: 
  - name: "[Vincent Gélinas, M.Sc.](https://compbio.ca/team/)"
  - name: "[Charles Joly Beauparlant, Ph.D](https://compbio.ca/teams/charles-joly-beauparlant/)"
date: last-modified
date-format: "DD-MM-YYYY"
title-block-banner: "#1A468F"
title-block-banner-color: "white"
description: "UMAPs, dotplots, and violin plots"
last-modified:
backgroundcolor: "white"
execute:
  echo: false
format:
  html:
    tabsets: true
    toc: true
    toc-depth: 3
    toc-title: Contents
    toc-location: left
    self-contained: true
    cards: true
    css: _report/adlab_html.css
    linkcolor: "#9DD3E4"
callout-icon: false
include-after-body:
  text: |
    &nbsp;
    <footer>
    <hr/>
    <p class="footer-information" style="text-align: center;">Produced by <a href="https://compbio.ca/">Arnaud Droit’s Computational Biology Laboratory</a></p>
     <p class="footer-information" style="text-align: center;"><span style="color: white;">Centre de recherche du CHU de Québec - Université Laval</span></p>
     <p class="footer-information" style="text-align: center;"><span style="color: white;">2705, boulevard Laurier, Québec (QC), G1V4G2, T-R5711</span></p>
    </p>
        &nbsp;

    </footer>
---

```{r libraries, echo = FALSE}
suppressMessages(library(ggplot2))
suppressMessages(library(patchwork))
suppressMessages(library(tidyverse))
suppressMessages(library(magrittr))
suppressMessages(library(dplyr))
suppressMessages(library(ggiraph))
suppressMessages(library(ggrastr))
suppressMessages(library(gridExtra))
suppressMessages(library(ComplexHeatmap))
suppressMessages(library(grid))
suppressMessages(library(knitr))
suppressMessages(library(DT))
suppressMessages(library(gprofiler2))
```

## Introduction

The goal of this report is to portrait the different gene expressions from a given gene list with graphical insights. 

These figures can help differentiate cell types between each other with clear gene expression patterns. 

## Methods

Analyses were performed using [Seurat](https://satijalab.org/seurat/) v*4.4.0 package [1]. For Uniform Manifold Approximation and Projection (UMAPs) and clusters computing, 20 PCA dimensions were used, and a series of resolution from 0.1 to 2 were chosen for clustering. All graphical representation were done using [ggplot2](https://ggplot2.tidyverse.org/reference/scale_brewer.html) v3.5.1 from the [tidyverse](https://www.tidyverse.org/) v2.0.0 package tool [2]. All R analyses were done in R v4.3.1 [3].

[1] Hao, Yuhan et al. Integrated analysis of multimodal single-cell data. Cell, Volume 184, Issue 13, 3573 - 3587.e29

[2] Wickham, H. (2011). ggplot2. Wiley Interdisciplinary Reviews: Computational
Statistics, 3(2), 180-185

[3] R: Team, R. Core. "R: A language and environment for statistical computing."
(2013): 201.

### Softwares

::: {.callout-note collapse=true .session-info}

## R session infos

```
R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Debian GNU/Linux 12 (bookworm)

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.21.so;  LAPACK version 3.11.0

locale:
 [1] LC_CTYPE=en_CA.UTF-8       LC_NUMERIC=C               LC_TIME=en_CA.UTF-8        LC_COLLATE=en_CA.UTF-8    
 [5] LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8    LC_PAPER=en_CA.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

time zone: Canada/Eastern
tzcode source: system (glibc)

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] DiffBind_3.10.1                           SummarizedExperiment_1.32.0              
 [3] MatrixGenerics_1.14.0                     matrixStats_1.5.0                        
 [5] rtracklayer_1.62.0                        org.Mm.eg.db_3.17.0                      
 [7] TxDb.Mmusculus.UCSC.mm9.knownGene_3.2.2   TxDb.Mmusculus.UCSC.mm10.knownGene_3.10.0
 [9] GenomicFeatures_1.52.2                    GenomicRanges_1.54.1                     
[11] GenomeInfoDb_1.38.8                       ggcorrplot_0.1.4.1                       
[13] corrplot_0.92                             RColorBrewer_1.1-3                       
[15] shinythemes_1.2.0                         bslib_0.8.0                              
[17] shiny_1.9.0                               org.Hs.eg.db_3.17.0                      
[19] AnnotationDbi_1.64.1                      IRanges_2.36.0                           
[21] S4Vectors_0.40.2                          Biobase_2.62.0                           
[23] BiocGenerics_0.48.1                       lubridate_1.9.2                          
[25] forcats_1.0.0                             stringr_1.5.1                            
[27] dplyr_1.1.4                               purrr_1.0.4                              
[29] readr_2.1.4                               tidyr_1.3.1                              
[31] tibble_3.2.1                              tidyverse_2.0.0                          
[33] ggplot2_3.4.4                            

loaded via a namespace (and not attached):
  [1] igraph_2.1.4                            ica_1.0-3                              
  [3] plotly_4.10.4                           devtools_2.4.5                         
  [5] zlibbioc_1.48.2                         tidyselect_1.2.1                       
  [7] bit_4.0.5                               lattice_0.21-8                         
  [9] SQUAREM_2021.1                          rjson_0.2.21                           
 [11] blob_1.2.4                              urlchecker_1.0.1                       
 [13] S4Arrays_1.2.1                          parallel_4.3.1                         
 [15] dichromat_2.0-0.1                       png_0.1-8                              
 [17] plotrix_3.8-2                           cli_3.6.5                              
 [19] ggplotify_0.1.2                         goftest_1.2-3                          
 [21] textshaping_0.3.6                       BiocIO_1.12.0                          
 [23] mixsqp_0.3-48                           uwot_0.1.16                            
 [25] shadowtext_0.1.2                        curl_5.2.1                             
 [27] mime_0.12                               evaluate_0.24.0                        
 [29] tidytree_0.4.5                          leiden_0.4.3                           
 [31] stringi_1.8.4                           backports_1.4.1                        
 [33] XML_3.99-0.15                           httpuv_1.6.15                          
 [35] magrittr_2.0.3                          rappdirs_0.3.3                         
 [37] splines_4.3.1                           jpeg_0.1-10                            
 [39] ggraph_2.1.0                            sctransform_0.4.1                      
 [41] ggbeeswarm_0.7.2                        sessioninfo_1.2.2                      
 [43] DBI_1.1.3                               corpcor_1.6.10                         
 [45] jquerylib_0.1.4                         genefilter_1.82.1                      
 [47] withr_3.0.2                             emdbook_1.3.13                         
 [49] systemfonts_1.0.4                       enrichplot_1.20.1                      
 [51] lmtest_0.9-40                           ggnewscale_0.4.9                       
 [53] clustree_0.5.1                          bdsmatrix_1.3-6                        
 [55] tidygraph_1.2.3                         sva_3.48.0                             
 [57] mixOmics_6.24.0                         htmlwidgets_1.6.4                      
 [59] fs_1.6.4                                biomaRt_2.56.1                         
 [61] ggrepel_0.9.4                           labeling_0.4.3                         
 [63] SparseArray_1.2.4                       DESeq2_1.40.2                          
 [65] cellranger_1.1.0                        annotate_1.78.0                        
 [67] truncnorm_1.0-9                         reticulate_1.32.0                      
 [69] zoo_1.8-12                              XVector_0.42.0                         
 [71] knitr_1.48                              timechange_0.2.0                       
 [73] apeglm_1.22.1                           patchwork_1.1.3                        
 [75] caTools_1.18.2                          grid_4.3.1                             
 [77] data.table_1.15.4                       ggtree_3.10.1                          
 [79] RSpectra_0.16-1                         irlba_2.3.5.1                          
 [81] ggrastr_1.0.2                           gridGraphics_0.5-1                     
 [83] ellipsis_0.3.2                          lazyeval_0.2.2                         
 [85] yaml_2.3.10                             survival_3.5-7                         
 [87] scattermore_1.2                         crayon_1.5.3                           
 [89] RcppAnnoy_0.0.21                        progressr_0.14.0                       
 [91] ChIPseeker_1.36.0                       tweenr_2.0.2                           
 [93] later_1.3.2                             ggridges_0.5.4                         
 [95] codetools_0.2-19                        profvis_0.3.8                          
 [97] Seurat_4.4.0                            KEGGREST_1.42.0                        
 [99] bbmle_1.0.25                            Rtsne_0.16                             
[101] limma_3.58.1                            Rsamtools_2.18.0                       
[103] filelock_1.0.2                          foreign_0.8-84                         
[105] pkgconfig_2.0.3                         xml2_1.3.5                             
[107] GenomicAlignments_1.38.0                aplot_0.2.0                            
[109] spatstat.sparse_3.0-2                   BSgenome_1.68.0                        
[111] ape_5.7-1                               viridisLite_0.4.2                      
[113] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2 xtable_1.8-4                           
[115] interp_1.1-4                            hwriter_1.3.2.1                        
[117] plyr_1.8.9                              ashr_2.2-63                            
[119] httr_1.4.7                              tools_4.3.1                            
[121] globals_0.16.2                          SeuratObject_4.1.4                     
[123] pkgbuild_1.4.4                          checkmate_2.3.0                        
[125] beeswarm_0.4.0                          nlme_3.1-162                           
[127] HDO.db_0.99.1                           dbplyr_2.3.3                           
[129] digest_0.6.36                           numDeriv_2016.8-1.1                    
[131] Matrix_1.6-1                            farver_2.1.2                           
[133] tzdb_0.4.0                              reshape2_1.4.4                         
[135] yulab.utils_0.0.9                       viridis_0.6.5                          
[137] glue_1.8.0                              cachem_1.1.0                           
[139] BiocFileCache_2.8.0                     polyclip_1.10-4                        
[141] generics_0.1.3                          Biostrings_2.70.3                      
[143] SingleCell_0.1.0                        mvtnorm_1.2-3                          
[145] amap_0.8-19                             parallelly_1.36.0                      
[147] pkgload_1.4.0                           statmod_1.5.0                          
[149] ragg_1.2.5                              pbapply_1.7-2                          
[151] vroom_1.6.4                             rio_0.5.30                             
[153] utf8_1.2.4                              invgamma_1.1                           
[155] graphlayouts_1.0.0                      gtools_3.9.4                           
[157] readxl_1.4.3                            gridExtra_2.3                          
[159] GenomeInfoDbData_1.2.11                 RCurl_1.98-1.13                        
[161] memoise_2.0.1                           rmarkdown_2.27                         
[163] pheatmap_1.0.12                         scales_1.4.0                           
[165] future_1.33.0                           RANN_2.6.1                             
[167] spatstat.data_3.0-1                     rstudioapi_0.17.1                      
[169] cluster_2.1.4                           spatstat.utils_3.0-3                   
[171] hms_1.1.3                               fitdistrplus_1.1-11                    
[173] cowplot_1.1.1                           ellipse_0.5.0                          
[175] colorspace_2.1-1                        rlang_1.1.6                            
[177] ggforce_0.4.1                           mgcv_1.8-42                            
[179] xfun_0.46                               coda_0.19-4                            
[181] rARPACK_0.11-0                          remotes_2.4.2.1                        
[183] abind_1.4-5                             GOSemSim_2.28.1                        
[185] treeio_1.26.0                           bitops_1.0-7                           
[187] promises_1.3.0                          scatterpie_0.2.1                       
[189] RSQLite_2.3.1                           qvalue_2.34.0                          
[191] openxlsx_4.2.5.2                        fgsea_1.28.0                           
[193] DelayedArray_0.28.0                     GO.db_3.18.0                           
[195] compiler_4.3.1                          prettyunits_1.2.0                      
[197] boot_1.3-28.1                           listenv_0.9.0                          
[199] Rcpp_1.0.13                             edgeR_4.0.16                           
[201] tensor_1.5                              usethis_2.2.2                          
[203] MASS_7.3-60                             progress_1.2.2                         
[205] BiocParallel_1.36.0                     spatstat.random_3.1-6                  
[207] R6_2.6.1                                fastmap_1.2.0                          
[209] fastmatch_1.1-4                         GreyListChIP_1.32.1                    
[211] vipor_0.4.5                             ROCR_1.0-11                            
[213] systemPipeR_2.6.3                       gtable_0.3.6                           
[215] KernSmooth_2.23-21                      latticeExtra_0.6-30                    
[217] miniUI_0.1.1.1                          deldir_1.0-9                           
[219] htmltools_0.5.8.1                       bit64_4.0.5                            
[221] spatstat.explore_3.2-3                  lifecycle_1.0.4                        
[223] zip_2.3.0                               restfulr_0.0.15                        
[225] sass_0.4.9                              vctrs_0.6.5                            
[227] spatstat.geom_3.2-5                     DOSE_3.28.2                            
[229] haven_2.5.3                             ggfun_0.1.2                            
[231] sp_2.0-0                                future.apply_1.11.0                    
[233] pillar_1.10.2                           gplots_3.1.3                           
[235] ShortRead_1.58.0                        locfit_1.5-9.8                         
[237] jsonlite_1.8.8

```
:::

## Markers

This a list of all the gene markers used in this report, and the cell type associated with: 

